webhooks-extension/pkg/endpoints/sink.go

```diff
@@ -299,6 +428,54 @@ func definePipelineRun(pipelineRunName, namespace, saName, repoURL string,
 	return pipelineRunPointer, nil
 }
 
+/* Create a new TaskRun - repoUrl, resourceBinding and params can be nill depending on the Pipeline
+each TaskRun has a 1 hour timeout: */
+func defineTaskRun(taskRunName, namespace, saName, repoURL string,
+	task v1alpha1.Task,
+	triggerType v1alpha1.TaskTriggerType,
+	resourceBinding []v1alpha1.TaskResourceBinding,
+	params []v1alpha1.Param) (*v1alpha1.TaskRun, error) {
+
+	gitServer, gitOrg, gitRepo := "", "", ""
+	err := errors.New("")
+	if repoURL != "" {
+		gitServer, gitOrg, gitRepo, err = getGitValues(repoURL)
+		if err != nil {
+			logging.Log.Errorf("error getting the Git values: %s.", err)
+			return &v1alpha1.TaskRun{}, err
+		}
+	}
+
+	taskRunData := v1alpha1.TaskRun{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      taskRunName,
+			Namespace: namespace,
+			Labels: map[string]string{
+				"app":          "tekton-webhook-handler",
+				gitServerLabel: gitServer,
+				gitOrgLabel:    gitOrg,
+				gitRepoLabel:   gitRepo,
```

Comment:
We want to add the branch label too I assume like I added for PipelineRuns

Reply:
I add it.  I'll rebase to make it compile.